packages/wonder-stuff-core/src/kind-error.js

```diff
@@ -0,0 +1,125 @@
+// @flow
+import {Errors} from "./errors.js";
+import {ErrorInfo} from "./error-info.js";
+import {cloneMetadata} from "./clone-metadata.js";
+
+import type {Metadata} from "./types.js";
+
+type Options = {|
+    cause?: Error,
+    metadata?: $ReadOnly<Metadata>,
+    prefix?: string,
+    stripStackFrames?: number,
+    minimumFrameCount?: number,
+|};
+
+/**
+ * An error to describe detailed states and relationships.
+ *
+ * This error type supports error taxonomies and attachment of metadata, as
+ * well as causal relationships between itself and another source error.
+ * More specific variations can be built off this to provide a payload for
+ * logging setups like Sentry or winston.
+ */
+export class KindError extends Error {
+    +kind: string;
+    +originalMessage: string;
+    +metadata: ?$ReadOnly<Metadata>;
+
+    /**
+     * Creates an instance of `KindError`.
+     *
+     * @param {string} message The error message.
+     * @param {TKinds} [kind] The kind of error. This will be combined with
+     * `prefix` to form the name of the error, i.e. PrefixKindError.
+     * Defaults to `Errors.Unknown`.
+     * @param {Error} [cause] The error that caused this error.
```

Comment:
Is the docstring syntax right here? Since you have an object, `Options`, as an arg, not the individual properties of the `Options` as args. I think for correct docstrings, you'd want to define the Options type, and doc it's props where you define the type.

Reply:
It might be fine... I'm not sure any more what kinds of fancy stuff is expected by documentation generators and IDEs.

Reply:
I haven't updated the docstring yet since I did the `Options` stuff.

Reply:
Updated the docstring now.